osu.Game/Modes/ScoreProcessor.cs

```diff
@@ -64,6 +78,12 @@ protected void TriggerFailed()
         {
             Failed?.Invoke();
         }
+
+        /// <summary>
+        /// Checks if the score is in a failing state.
+        /// </summary>
+        /// <returns>Whether the score is in a failing state.</returns>
+        public abstract bool CheckFailed();

```

Comment:
i feel like this should be private `updateFailed`, containing the trigger logic.

then you can have a protected or public
`virtual HasFailed => false;` which has the actual check logic.

reasoning: it is internally *potentially* triggering something, not just checking. it could lead to scenarios we didn't expect if called from outside.